src/main.rs

```diff
@@ -53,32 +53,44 @@ pub fn search_mods(query: &String) -> Result<MinecraftMods, std::io::Error> {
     Ok(mods)
 }
 
-pub fn find_correct_version(id: &String, target: &Version) -> Result<ModFile, std::io::Error> {
-    let url = format!("https://api.modrinth.com/api/v1/mod/{}/version", id);
-    let versions: Vec<ModVersion> = get(url)
+pub fn find_correct_version(id: &String, target: &Version) -> Result<ModFile, Error> {
+    let url: String = format!("https://api.modrinth.com/api/v1/mod/{}/version", id);
+    let mod_versions: Vec<ModVersion> = get(url)
         .send()?
         .json()?;
 
-    for version in versions {
-        if !version.loaders.contains(&"fabric".into()) {
-            continue
-        }
-        let found = version
-            .versions
+    for mod_version in mod_versions {
+        if !mod_version.loaders.contains(&"fabric".into()) { continue; }
+
+        let found_version_index: Option<usize> = mod_version
+            .game_versions
             .iter()
-            .any(|ver| match VersionReq::parse(ver) {
-                Ok(ver) => ver.matches(&target),
-                Err(_) => false,
+            .position(|ver| {
+                let attempt = VersionReq::parse(format!("={}", ver).as_str());
+                match attempt {
+                    Ok(parsed_ver) => parsed_ver.matches(&target),
+                    Err(_) => false
+                }
             });
-        if found {
-            return Ok(version.files.get(0).unwrap().to_owned());
+
+        match found_version_index {
+            Some(idx) => { return Ok(mod_version.files[idx].to_owned()); },
+            None => { continue; }
         }
     }
- 
-    Err(std::io::Error::new(std::io::ErrorKind::NotFound, "cant find correct version"))
+
+    Err(
+        Error::new(
+            ErrorKind::NotFound,
+            format!(
+                "could not find version that satisfies target {} and loader paper",
```

Comment:
> loader paper

Reply:
planning to add forge loader support plus loader json field later on

Reply:
wait
im dumb